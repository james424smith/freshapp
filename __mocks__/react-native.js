import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
