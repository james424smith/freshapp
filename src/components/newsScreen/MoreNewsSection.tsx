import React from "react";
import { View, TouchableOpacity } from "react-native";
import Text from "../StyledText";
import { Newsletters, PressReleases, NewsType } from "../../interfaces";
import styles from "../../styles/more_screens/NewsCardStyles";
import { READ_MORE_NEWS } from "../../constants/NewsConstants";
import { useTheme } from "@react-navigation/native";

type Props = {
  title: string;
  summary: string;
  navigate: (arg1: string, arg2?: unknown) => void;
  newItemType: NewsType;
  newItem: Newsletters | PressReleases;
};

const MoreNewsSection = (props: Props) => {
  const { navigate, newItem, newItemType, summary, title } = props;
  const theme = useTheme();

  const navigateToReadMore = () => {
    navigate(READ_MORE_NEWS, {
      newItem: {
        ...newItem,
        newItemType: newItemType,
      },
    });
  };

  return (
    <View
      style={{
        ...styles.prenewsListContainer,
        backgroundColor: theme.colors.background,
      }}
    >
      <View
        style={{
          ...styles.opacityStyle,
          backgroundColor: theme.colors.primary,
        }}
      >
        <TouchableOpacity
          testID="news-button"
          onPress={() => navigateToReadMore()}
          style={styles.textMargin}
        >
          <Text style={{ ...styles.prenewsHeader, color: theme.colors.text }}>
            {title}
          </Text>
          <Text style={{ ...styles.prenewssummary, color: theme.colors.text }}>
            {summary}
          </Text>
          <TouchableOpacity
            testID={"readmore-button"}
            onPress={() => navigateToReadMore()}
          >
            <Text
              style={{
                ...styles.readMoreText,
                color: theme.colors.newAquaMarineColor,
              }}
            >
              Read more
            </Text>
          </TouchableOpacity>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default MoreNewsSection;
