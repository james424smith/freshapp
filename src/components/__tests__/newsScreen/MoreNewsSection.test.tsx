import React from "react";
import MoreNewsSection from "../../newsScreen/MoreNewsSection";
import { render, cleanup, fireEvent } from "@testing-library/react-native";

describe("MoreNewsSection snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correct ui", () => {
    const navigation = jest.fn();

    const { toJSON } = render(
      <MoreNewsSection
        title="title"
        summary="summary"
        navigate={navigation}
        newItemType={"Newsletters"}
        newItem={{
          document: "1234",
          title: "title",
          summary: "",
        }}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});

describe("firevents tests", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });
  it("should call navigate when news-button is pressed", () => {
    const navigation = jest.fn();

    const { getByTestId } = render(
      <MoreNewsSection
        title="title"
        summary="summary"
        navigate={navigation}
        newItemType={"Newsletters"}
        newItem={{
          document: "1234",
          title: "title",
          summary: "",
        }}
      />
    );
    fireEvent.press(getByTestId("news-button"));
    expect(navigation).toHaveBeenCalledWith("ReadMoreNews", {
      newItem: {
        document: "1234",
        title: "title",
        summary: "",
        newItemType: "Newsletters",
      },
    });
  });
  it("should call navigate when readmore-button is pressed", () => {
    const navigation = jest.fn();

    const { getByTestId } = render(
      <MoreNewsSection
        title="title"
        summary="summary"
        navigate={navigation}
        newItemType={"Newsletters"}
        newItem={{
          document: "1234",
          title: "title",
          summary: "",
        }}
      />
    );
    fireEvent.press(getByTestId("readmore-button"));
    expect(navigation).toHaveBeenCalledWith("ReadMoreNews", {
      newItem: {
        document: "1234",
        title: "title",
        summary: "",
        newItemType: "Newsletters",
      },
    });
  });
});
