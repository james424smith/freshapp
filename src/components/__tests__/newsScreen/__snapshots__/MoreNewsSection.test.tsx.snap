// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreNewsSection snapshot test should render correct ui 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(242, 242, 242)",
      "flex": 1,
      "marginLeft": 26.5,
      "marginRight": 26.5,
      "marginTop": 37.5,
      "width": "93%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(0, 122, 255)",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "borderTopLeftRadius": 5,
        "borderTopRightRadius": 5,
        "elevation": 5,
        "flex": 1,
        "shadowColor": "#78849E",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "margin": 37.5,
          "opacity": 1,
        }
      }
      testID="news-button"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": "rgb(28, 28, 30)",
              "fontSize": 29.5,
              "fontWeight": "bold",
              "marginTop": 16,
              "paddingBottom": 15,
            },
          ]
        }
      >
        title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            Object {
              "color": "rgb(28, 28, 30)",
              "fontSize": 26.5,
              "marginBottom": 30,
            },
          ]
        }
      >
        summary
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="readmore-button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": undefined,
                "fontSize": 26.5,
                "marginTop": 37.5,
                "paddingRight": 30,
                "textAlign": "right",
              },
            ]
          }
        >
          Read more
        </Text>
      </View>
    </View>
  </View>
</View>
`;
